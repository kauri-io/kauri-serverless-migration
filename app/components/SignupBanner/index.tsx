import styled from 'styled-components'
import { Title1, BodyCard } from '../Typography'
import Button from '../../components/Button'
import Stack from 'stack-styled'
import Link from 'next/link'

const ResourceDetailsContainer = styled.section`
    display: flex;
    flex-direction: column;
    max-width: 700px;
    > :not(:last-child) {
        margin-bottom: ${props => props.theme.space[2]}px;
    }
`

const ViewContainer = styled.section`
    display: flex;
    flex-direction: row;
    margin-left: auto;
    > :first-child {
        margin-right: ${props => props.theme.space[2]}px;
    }
    @media (max-width: ${props => props.theme.breakpoints[0]}) {
        display: none;
    }
`

const SignupBannerStack = styled(Stack)`
    padding: ${props => props.theme.space[3]}px ${props => props.theme.padding};
    background: ${props => props.theme.colors.bgPrimary};
    @media (max-width: ${props => props.theme.breakpoints[0]}) {
        padding: ${props => props.theme.space[3]}px
            ${props => props.theme.space[2]}px;
    }
`

const SignupBanner: React.FunctionComponent = () => (
    <SignupBannerStack
        alignItems={['']}
        justifyContent={['']}
        gridAutoFlow={['column']}
        gap={30}
    >
        <ResourceDetailsContainer>
            <Title1 color="white">Learn to Build on Ethereum with Kauri</Title1>
            <BodyCard color="white">
                Articles, Tutorials, Documentation and Best Practices
            </BodyCard>
        </ResourceDetailsContainer>
        <ViewContainer>
            <Link href="/login">
                <a>
                    <Button
                        color="primary"
                        variant="contained"
                    >{`Sign up`}</Button>
                </a>
            </Link>
            <Link href="/help">
                <a>
                    <Button color="primary" variant="outlined">
                        Learn about kauri
                    </Button>
                </a>
            </Link>
        </ViewContainer>
    </SignupBannerStack>
)

export default SignupBanner
