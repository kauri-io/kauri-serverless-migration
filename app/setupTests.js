import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import theme from './lib/theme-config'
import { ThemeProvider } from 'styled-components'

Enzyme.configure({ adapter: new Adapter() })

const observeMock = {
    observe: () => null,
    disconnect: () => null, // maybe not needed
}

window.IntersectionObserver = () => observeMock

export const mountWithTheme = children =>
    mount(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const renderWithTheme = children =>
    render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const shallowWithTheme = children =>
    shallow(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
