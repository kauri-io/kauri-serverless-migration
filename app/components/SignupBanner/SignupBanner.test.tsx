import SignupBanner from './index'
import { mountWithTheme } from '../../setupTests'

describe('components/SignupBanner', () => {
    it('should match snapshot', () => {
        const wrapper = mountWithTheme(<SignupBanner />)
        expect(wrapper).toMatchSnapshot()
    })
})
