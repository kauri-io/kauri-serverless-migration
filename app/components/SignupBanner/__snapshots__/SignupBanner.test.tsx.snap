// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/SignupBanner should match snapshot 1`] = `
<ThemeProvider
  theme={
    Object {
      "DEFAULT_CARD_WIDTH": 288,
      "MAIN_BODY_WIDTH": 1242,
      "bg": Object {
        "bgPrimary": "#1E2428",
        "bgSecondary": "#1E3D3B",
        "borderTextArea": "#979797",
        "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
        "disabledBackgroundColor": "#D6D6D6",
        "disabledTextColor": "#9B9B9B",
        "divider": "#EBEBEB",
        "errorRedColor": "#C03030",
        "hoverTextColor": "#108E72",
        "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
        "paddingTop": "2.5em",
        "primary": "#0BA986",
        "primaryColor": "#0BA986",
        "primaryDark": "#11856B",
        "primaryTextColor": "#1E2428",
        "secondary": "#11856B",
        "secondaryBlue": "rgb(81, 123, 162, 1)",
        "secondaryBlueDark": "rgb(81, 118, 142, 1)",
        "secondaryColor": "#1E3D3B",
        "secondaryTextColor": "#283035",
        "tertiaryBackgroundColor": "#F7F7F7",
        "textPrimary": "#1E2428",
        "white": "#FFFFFF",
      },
      "bgPrimary": "#1E2428",
      "bgSecondary": "#1E3D3B",
      "borderTextArea": "#979797",
      "breakpoints": Array [
        "500px",
        "52em",
        "1242px",
      ],
      "colors": Object {
        "bgPrimary": "#1E2428",
        "bgSecondary": "#1E3D3B",
        "borderTextArea": "#979797",
        "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
        "disabledBackgroundColor": "#D6D6D6",
        "disabledTextColor": "#9B9B9B",
        "divider": "#EBEBEB",
        "errorRedColor": "#C03030",
        "hoverTextColor": "#108E72",
        "padding": "calc((100vw - \${props => props.theme.breakpoints[2]}) / 2)",
        "paddingTop": "2.5em",
        "primary": "#0BA986",
        "primaryColor": "#0BA986",
        "primaryDark": "#11856B",
        "primaryTextColor": "#1E2428",
        "secondary": "#11856B",
        "secondaryBlue": "rgb(81, 123, 162, 1)",
        "secondaryBlueDark": "rgb(81, 118, 142, 1)",
        "secondaryColor": "#1E3D3B",
        "secondaryTextColor": "#283035",
        "tertiaryBackgroundColor": "#F7F7F7",
        "textPrimary": "#1E2428",
        "white": "#FFFFFF",
      },
      "contentBorder": "1px solid rgba(30, 36, 40, 0.19)",
      "disabledBackgroundColor": "#D6D6D6",
      "disabledTextColor": "#9B9B9B",
      "divider": "#EBEBEB",
      "errorRedColor": "#C03030",
      "fontSizes": Array [
        11,
        13,
        14,
        16,
        18,
        20,
        22,
        28,
        32,
      ],
      "fontWeight": Array [
        300,
        "normal",
        500,
        700,
      ],
      "hoverTextColor": "#108E72",
      "padContent": "
    max-width: 1242px;
    margin-left: auto;
    margin-right: auto;
    @media (min-width: 1242px) {
      min-width: 1242px;
    }
    @media (max-width: 1242px) {
      min-width: 90%;
    }
  ",
      "padding": "calc((100vw - 1242px) / 2)",
      "paddingTop": "2.5em",
      "primary": "#0BA986",
      "primaryColor": "#0BA986",
      "primaryDark": "#11856B",
      "primaryTextColor": "#1E2428",
      "secondary": "#11856B",
      "secondaryBlue": "rgb(81, 123, 162, 1)",
      "secondaryBlueDark": "rgb(81, 118, 142, 1)",
      "secondaryColor": "#1E3D3B",
      "secondaryTextColor": "#283035",
      "space": Array [
        0,
        10,
        15,
        30,
        60,
      ],
      "tertiaryBackgroundColor": "#F7F7F7",
      "textPrimary": "#1E2428",
      "white": "#FFFFFF",
    }
  }
>
  <SignupBanner>
    <SignupBanner__SignupBannerStack
      alignItems={
        Array [
          "",
        ]
      }
      gap={30}
      gridAutoFlow={
        Array [
          "column",
        ]
      }
      justifyContent={
        Array [
          "",
        ]
      }
    >
      <StyledComponent
        alignItems={
          Array [
            "",
          ]
        }
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "_foldedDefaultProps": Object {
              "gap": 2,
            },
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "SignupBanner__SignupBannerStack-sc-1hxzf8p-2",
              "isStatic": false,
              "lastClassName": "ieQzri",
              "rules": Array [
                "box-sizing: border-box;",
                "display: grid;",
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                "padding:",
                [Function],
                "px ",
                [Function],
                ";background:",
                [Function],
                ";@media (max-width:",
                [Function],
                "){padding:",
                [Function],
                "px ",
                [Function],
                "px;}",
              ],
            },
            "displayName": "SignupBanner__SignupBannerStack",
            "foldedComponentIds": Array [
              "sc-bdVaJa",
            ],
            "propTypes": Object {
              "alignContent": [Function],
              "alignItems": [Function],
              "alignSelf": [Function],
              "display": [Function],
              "flex": [Function],
              "flexBasis": [Function],
              "flexDirection": [Function],
              "flexGrow": [Function],
              "flexShrink": [Function],
              "flexWrap": [Function],
              "gap": [Function],
              "gridArea": [Function],
              "gridAutoColumns": [Function],
              "gridAutoFlow": [Function],
              "gridAutoRows": [Function],
              "gridColumn": [Function],
              "gridColumnGap": [Function],
              "gridGap": [Function],
              "gridRow": [Function],
              "gridRowGap": [Function],
              "gridTemplateAreas": [Function],
              "gridTemplateColumns": [Function],
              "gridTemplateRows": [Function],
              "height": [Function],
              "justifyContent": [Function],
              "justifyItems": [Function],
              "justifySelf": [Function],
              "m": [Function],
              "margin": [Function],
              "marginBottom": [Function],
              "marginLeft": [Function],
              "marginRight": [Function],
              "marginTop": [Function],
              "marginX": [Function],
              "marginY": [Function],
              "maxHeight": [Function],
              "maxWidth": [Function],
              "mb": [Function],
              "minHeight": [Function],
              "minWidth": [Function],
              "ml": [Function],
              "mr": [Function],
              "mt": [Function],
              "mx": [Function],
              "my": [Function],
              "order": [Function],
              "overflow": [Function],
              "p": [Function],
              "padding": [Function],
              "paddingBottom": [Function],
              "paddingLeft": [Function],
              "paddingRight": [Function],
              "paddingTop": [Function],
              "paddingX": [Function],
              "paddingY": [Function],
              "pb": [Function],
              "pl": [Function],
              "pr": [Function],
              "pt": [Function],
              "px": [Function],
              "py": [Function],
              "size": [Function],
              "verticalAlign": [Function],
              "width": [Function],
            },
            "render": [Function],
            "styledComponentId": "SignupBanner__SignupBannerStack-sc-1hxzf8p-2",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        gap={30}
        gridAutoFlow={
          Array [
            "column",
          ]
        }
        justifyContent={
          Array [
            "",
          ]
        }
      >
        <div
          className="sc-bdVaJa SignupBanner__SignupBannerStack-sc-1hxzf8p-2 ieQzri"
        >
          <SignupBanner__ResourceDetailsContainer>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "SignupBanner__ResourceDetailsContainer-sc-1hxzf8p-0",
                    "isStatic": false,
                    "lastClassName": "jkhksa",
                    "rules": Array [
                      "display:flex;flex-direction:column;max-width:700px;>:not(:last-child){margin-bottom:",
                      [Function],
                      "px;}",
                    ],
                  },
                  "displayName": "SignupBanner__ResourceDetailsContainer",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "SignupBanner__ResourceDetailsContainer-sc-1hxzf8p-0",
                  "target": "section",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <section
                className="SignupBanner__ResourceDetailsContainer-sc-1hxzf8p-0 jkhksa"
              >
                <Typography__styledComponent
                  color="white"
                >
                  <StyledComponent
                    color="white"
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "Typography__styledComponent-sc-36xkf6-5",
                          "isStatic": false,
                          "lastClassName": "iLJkHN",
                          "rules": Array [
                            "color:",
                            [Function],
                            ";margin:0px;font-weight:",
                            "500",
                            ";",
                            ";font-size:",
                            "28",
                            "px;text-transform:",
                            [Function],
                            ";text-align:",
                            [Function],
                            ";:hover{color:",
                            [Function],
                            ";cursor:",
                            [Function],
                            ";}",
                          ],
                        },
                        "displayName": "Typography__styledComponent",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "Typography__styledComponent-sc-36xkf6-5",
                        "target": "span",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <span
                      className="Typography__styledComponent-sc-36xkf6-5 iLJkHN"
                      color="white"
                    >
                      Learn to Build on Ethereum with Kauri
                    </span>
                  </StyledComponent>
                </Typography__styledComponent>
                <Typography__component
                  color="white"
                >
                  <StyledComponent
                    color="white"
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "Typography__component-sc-36xkf6-4",
                          "isStatic": false,
                          "lastClassName": "dvuWvG",
                          "rules": Array [
                            "font-size:14px;font-weight:normal;letter-spacing:-0.1px;margin-bottom:",
                            [Function],
                            "px;line-height:18px;color:",
                            [Function],
                            ";",
                            ";",
                            [Function],
                            ";",
                          ],
                        },
                        "displayName": "Typography__component",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "Typography__component-sc-36xkf6-4",
                        "target": "span",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <span
                      className="Typography__component-sc-36xkf6-4 dvuWvG"
                      color="white"
                    >
                      Articles, Tutorials, Documentation and Best Practices
                    </span>
                  </StyledComponent>
                </Typography__component>
              </section>
            </StyledComponent>
          </SignupBanner__ResourceDetailsContainer>
          <SignupBanner__ViewContainer>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "SignupBanner__ViewContainer-sc-1hxzf8p-1",
                    "isStatic": false,
                    "lastClassName": "iebiGj",
                    "rules": Array [
                      "display:flex;flex-direction:row;margin-left:auto;>:first-child{margin-right:",
                      [Function],
                      "px;}@media (max-width:",
                      [Function],
                      "){display:none;}",
                    ],
                  },
                  "displayName": "SignupBanner__ViewContainer",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "SignupBanner__ViewContainer-sc-1hxzf8p-1",
                  "target": "section",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <section
                className="SignupBanner__ViewContainer-sc-1hxzf8p-1 iebiGj"
              >
                <Link
                  href="/login"
                  prefetch={true}
                >
                  <a
                    href="/login"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                  >
                    <PrimaryButtonComponent
                      color="white"
                    >
                      <PrimaryButton
                        bg="primary"
                        color="white"
                        fontSize={0}
                        fontWeight={700}
                        mr={2}
                        type="submit"
                      >
                        <StyledComponent
                          bg="primary"
                          color="white"
                          fontSize={0}
                          fontWeight={700}
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "PrimaryButton-sc-1cbxa53-0",
                                "isStatic": false,
                                "lastClassName": "bVbdII",
                                "rules": Array [
                                  "display:flex;align-items:center;justify-content:center;min-width:136px;max-width:150px;",
                                  [Function],
                                  ";height:40px;border-radius:4px;border:none;cursor:pointer;> svg,img{height:16px;",
                                  [Function],
                                  ";}text-transform:uppercase;opacity:",
                                  [Function],
                                  ";",
                                  [Function],
                                  ";",
                                  [Function],
                                  ";",
                                  [Function],
                                  ";",
                                  [Function],
                                  ";:hover{cursor:",
                                  [Function],
                                  ";}",
                                  ";:hover{background-color:",
                                  [Function],
                                  ";}",
                                ],
                              },
                              "displayName": "PrimaryButton",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "PrimaryButton-sc-1cbxa53-0",
                              "target": "button",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                          mr={2}
                          type="submit"
                        >
                          <button
                            className="PrimaryButton-sc-1cbxa53-0 bVbdII"
                            color="white"
                            fontSize={0}
                            fontWeight={700}
                            type="submit"
                          >
                            Sign up
                          </button>
                        </StyledComponent>
                      </PrimaryButton>
                    </PrimaryButtonComponent>
                  </a>
                </Link>
                <Link
                  href="/help"
                  prefetch={true}
                >
                  <a
                    href="/help"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                  >
                    <SecondaryButtonComponent
                      border="white"
                      color="white"
                    >
                      <SecondaryButton
                        bg="transparent"
                        border="white"
                        borderHover="primary"
                        color="white"
                        fontSize={0}
                        fontWeight={700}
                        mr={2}
                        type="button"
                      >
                        <StyledComponent
                          bg="transparent"
                          border="white"
                          borderHover="primary"
                          color="white"
                          fontSize={0}
                          fontWeight={700}
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "SecondaryButton-sy4hb2-0",
                                "isStatic": false,
                                "lastClassName": "fCmNwl",
                                "rules": Array [
                                  "display:flex;align-items:center;justify-content:center;min-width:136px;max-width:150px;",
                                  [Function],
                                  ";height:40px;border-radius:4px;border:none;cursor:pointer;> svg,img{height:16px;",
                                  [Function],
                                  ";}text-transform:uppercase;opacity:",
                                  [Function],
                                  ";",
                                  [Function],
                                  ";",
                                  [Function],
                                  ";",
                                  [Function],
                                  ";",
                                  [Function],
                                  ";:hover{cursor:",
                                  [Function],
                                  ";}",
                                  ";box-shadow:0px 0px 0px 1px ",
                                  [Function],
                                  ";:hover{box-shadow:0px 0px 0px 2px ",
                                  [Function],
                                  ";}",
                                ],
                              },
                              "displayName": "SecondaryButton",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "SecondaryButton-sy4hb2-0",
                              "target": "button",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                          mr={2}
                          type="button"
                        >
                          <button
                            className="SecondaryButton-sy4hb2-0 fCmNwl"
                            color="white"
                            fontSize={0}
                            fontWeight={700}
                            type="button"
                          >
                            Learn more
                          </button>
                        </StyledComponent>
                      </SecondaryButton>
                    </SecondaryButtonComponent>
                  </a>
                </Link>
              </section>
            </StyledComponent>
          </SignupBanner__ViewContainer>
        </div>
      </StyledComponent>
    </SignupBanner__SignupBannerStack>
  </SignupBanner>
</ThemeProvider>
`;
